replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/modern/ReplicationPersistentStoreImpl.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.modern;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ditto.replication.api.modern.Persistable;
+import org.codice.ditto.replication.api.modern.ReplicationPersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReplicationPersistentStoreImpl implements ReplicationPersistentStore {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(ReplicationPersistentStoreImpl.class);
+
+  private static final int DEFAULT_PAGE_SIZE = 100;
+
+  private static final int DEFAULT_START_INDEX = 0;
+
+  private PersistentStore persistentStore;
+
+  public ReplicationPersistentStoreImpl(PersistentStore persistentStore) {
+    this.persistentStore = persistentStore;
+  }
+
+  @Override
+  public <T extends Persistable> Optional<T> get(Class<T> klass, String typeName, String id) {
+    String cql = String.format("'id' = '%s'", id);
+    List<Map<String, Object>> storedMaps = Collections.emptyList();
+    try {
+      storedMaps = persistentStore.get(typeName, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+    } catch (PersistenceException e) {
+      LOGGER.debug(
+          "An error occurred while retrieving a {} with an ID of {}", klass.getName(), id, e);
+    }
+
+    if (storedMaps.isEmpty()) {
+      LOGGER.debug("Couldn't find persisted object of type {} with ID {}", typeName, id);
+      return Optional.empty();
+    } else {
+      if (storedMaps.size() > 1) {
+        LOGGER.debug("Found two objects in persistence of type {} with ID {}", typeName, id);
+      }
+      return Optional.of(Persistable.createPersistable(klass, storedMaps.get(0)));
+    }
+  }
+
+  @Override
+  public <T extends Persistable> List<T> get(Class<T> klass, String typeName) {
+    List<Map<String, Object>> storedMaps = Collections.emptyList();
+    try {
+      storedMaps = persistentStore.get(typeName, "", DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+    } catch (PersistenceException e) {
+      LOGGER.debug("An error occurred while retrieving data of type {}.", typeName, e);
```

Comment:
❓ Why not returned directly with an `Collections.emptyList()` or an empty list?

The whole method could be simplified as:

```
try {
    try {
      return persistentStore.get(typeName, "", DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE)
        .stream()
        .map(m -> Persistable.createPersistable(kclass, m))
        .collect(Collectors.toList());
    } catch (PersistenceException e) {
      LOGGER.debug("An error occurred while retrieving data of type {}.", typeName, e);
      return new ArrayList<>(); // or Collections.emptyList();
   }
```

Same comment as above, exceptions should not result in an empty result. Empty means there was nothing in the persistence store whereas as an error is an indication that you cannot tell. The client should be handling that on its own.